From de893f1b7bc9c2f221b102145e1a15f4b2521954 Mon Sep 17 00:00:00 2001
From: Ivan Murashov <ivan.murashov@lge.com>
Date: Tue, 21 Jun 2022 19:29:30 +0000
Subject: [PATCH] [Cast Streaming] libstdc++: Fix incomplete type of
 cast_streaming::ResourceProvider

Destructor of std::unique_ptr in libstdc++ uses sizeof() which requires
full definition of cast_streaming::ResourceProvider for return type of
ContentRendererClient::CreateCastStreamingResourceProvider().
Added required include and deps to fix the issue.

Bug: b/228081914, 957519
Change-Id: I1f12d104a090460f5d2ffbb78c72f59279a63b33
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3672799
Commit-Queue: Ivan Murashov <ivan.murashov@lge.com>
Reviewed-by: Jordan Bayles <jophba@chromium.org>
Reviewed-by: Fabrice de Gans <fdegans@chromium.org>
Reviewed-by: Alex Moshchuk <alexmos@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1016345}
---
 content/public/renderer/BUILD.gn                   | 1 +
 content/public/renderer/DEPS                       | 1 +
 content/public/renderer/content_renderer_client.cc | 1 +
 3 files changed, 3 insertions(+)

diff --git a/content/public/renderer/BUILD.gn b/content/public/renderer/BUILD.gn
index 1eab7f35738a7..fecca01152929 100644
--- a/content/public/renderer/BUILD.gn
+++ b/content/public/renderer/BUILD.gn
@@ -58,6 +58,7 @@ target(link_target_type, "renderer_sources") {
   ]
 
   deps = [
+    "//components/cast_streaming/renderer:resource_provider",
     "//content/public/child:child_sources",
     "//content/public/common:common_sources",
     "//content/renderer",
diff --git a/content/public/renderer/DEPS b/content/public/renderer/DEPS
index 12f044c8b6dbf..9e762d37425d9 100644
--- a/content/public/renderer/DEPS
+++ b/content/public/renderer/DEPS
@@ -1,4 +1,5 @@
 include_rules = [
+  "+components/cast_streaming/renderer/public",
   "+content/common/media",
   "+content/public/child",
   "+gin",
diff --git a/content/public/renderer/content_renderer_client.cc b/content/public/renderer/content_renderer_client.cc
index de7f7113f06c9..8eeeee0178f52 100644
--- a/content/public/renderer/content_renderer_client.cc
+++ b/content/public/renderer/content_renderer_client.cc
@@ -6,6 +6,7 @@
 
 #include "base/command_line.h"
 #include "build/build_config.h"
+#include "components/cast_streaming/renderer/public/resource_provider.h"
 #include "content/public/common/content_switches.h"
 #include "media/base/demuxer.h"
 #include "media/base/renderer_factory.h"
